### ISSUE TITLE
Formatting of lists inside documentation comments

### ISSUE THREAD

**Issue by diml** [ocaml-ppx/ocamlformat#477]
I'm trying the recently added support for formatting documentation comments and I find the formatting of lists a little bit too compact. For instance, for the following input:

```ocaml
(** list with short lines:

    - x
    - y
    - z

    list with long lines:

    - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    - yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy
    - zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz

    enumerated list with long lines:

    {ol
     {- xxxxxxxxxxxxxxxxxxxxxxx}
     {- yyyyyyyyyyyyyyyyyyyyyyy}
     {- zzzzzzzzzzzzzzzzzzzzzzz}
    }
*)
```

ocamlformat (at 9d23a512b160c0c78a27616b9d3ddb34d8333451) prints:

```ocaml
(** list with short lines:

    {ul {- x} {- y} {- z}} list with long lines:

    {ul {- xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx}
          {- yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy}
          {- zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz}} enumerated list with long
    lines:

    {ol {- xxxxxxxxxxxxxxxxxxxxxxx} {- yyyyyyyyyyyyyyyyyyyyyyy}
          {- zzzzzzzzzzzzzzzzzzzzzzz}} *)
```

It would seem nicer to me if the output was more something like this:

```ocaml
(** list with short lines:

    {ul
     {- x}
     {- y}
     {- z}}

    list with long lines:

    {ul
     {- xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx}
     {- yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy}
     {- zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz}}

    enumerated list with long lines:

    {ol
     {- xxxxxxxxxxxxxxxxxxxxxxx}
     {- yyyyyyyyyyyyyyyyyyyyyyy}
     {- zzzzzzzzzzzzzzzzzzzzzzz}} *)
```

i.e. with each item on its own line and with the same indentation. In particular this is closer to the output of ocamldoc/odoc. I'm happy to send a PR to implement this.

**Comment by jberdine** [comment]
I think that aiming for the same formatting as ocamldoc/odoc is the right goal.

I also wonder what others' view is on trying to emit output with as little markup as possible? For example, would it make sense to try to leave out `{ul`, etc. in the example above, as long as it parses back to the same value?

**Comment by jberdine** [comment]
To be more precise: I think it would be an improvement to format `ul` lists which have no sublists using the plain `-` shortcut, and `ol` lists without sublists using the `+` shortcut.

**Comment by diml** [comment]
I think so as well, the lighter syntax looks nicer.

**Comment by gpetiot** [comment]
Closed with #480
